core/chains/evm/client/rpc_client_test.go

```diff
@@ -160,33 +156,13 @@ func TestRPCClient_SubscribeNewHead(t *testing.T) {
 		assert.Equal(t, (*big.Int)(nil), highestUserObservations.TotalDifficulty)
 	})
 	t.Run("SubscribeToHeads with http polling enabled will update new heads", func(t *testing.T) {
-		type rpcServer struct {
```

Comment:
Why this change?
Could you also go through the rpc_client_tests and make sure that `NodeNewHeadsPollInterval` and `NodeFinalizedBlockPollInterval` are properly set (as originally specified)? Otherwise we won't have proper coverage for polling functionality.

Reply:
Also could you check that there are no duplicate test cases due to the removal of `SubscribeNewHead`

Reply:
Seems like `NodeNewHeadsPollInterval ` was actually disabled for the tests. When enabling it, we can't use this server since polling will be used. I modified the test server for this to enable passing heads to poll for and got things working as expected now. Most tests now use the head polling, but also have some using WS.